test/test_autotag.py

```diff
@@ -1040,3 +1045,35 @@ def test_ampersand_expansion(self):
     def test_accented_characters(self):
         dist = string_dist("\xe9\xe1\xf1", "ean")
         assert dist == 0.0
+
+
+@pytest.mark.parametrize(
+    "single_field,list_field",
+    [
+        ("mb_artistid", "mb_artistids"),
+        ("mb_albumartistid", "mb_albumartistids"),
+        ("albumtype", "albumtypes"),
+    ],
+)
+@pytest.mark.parametrize(
+    "single_value,list_value",
+    [
+        (None, []),
+        (None, ["1"]),
+        (None, ["1", "2"]),
+        ("1", []),
+        ("1", ["1"]),
+        ("1", ["1", "2"]),
+        ("1", ["2", "1"]),
+    ],
+)
```

Comment:
and this!

Reply:
I'd like to augment this data with the data I find for albumtype in my (possibly broken) dev-library, but I need to really understand it first. Thanks!

Reply:
Well I'm not 100% sure what I'm doing but I think I understood roughly what this is doing.

I tried this with mouse-copy/pasted data from my terminal output.

```
@pytest.mark.parametrize(
    "single_value,list_value",
    [
        (None, []),
        (None, ["1"]),
        (None, ["1", "2"]),
        ("1", []),
        ("1", ["1"]),
        ("1", ["1", "2"]),
        ("1", ["2", "1"]),
        ('12", 33 ⅓ RPM, EP', ['1', '12", 33 ⅓ RPM, EP']),
        ('12", 33 ⅓ RPM, EP', ['12", 33 ⅓ RPM, EP', 2]),
    ],
)
```

```
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value7-mb_artistid-mb_artistids] PASSED                                                       [22/27]
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value7-mb_albumartistid-mb_albumartistids] PASSED                                             [23/27]
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value7-albumtype-albumtypes] PASSED                                                           [24/27]
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value8-mb_artistid-mb_artistids] PASSED                                                       [25/27]
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value8-mb_albumartistid-mb_albumartistids] PASSED                                             [26/27]
test/test_autotag.py::test_ensure_consistent_list_fields[12", 33 \u2153 RPM, EP-list_value8-albumtype-albumtypes] PASSED                                                           [27/27]
```

Reply:
That's the test cases for `single_field` (like `mb_artistid`) and `list_field` (like `mb_artistids`), covering cases where `mb_artistid` is not set, or `mb_artistids` is missing, or where first item in `mb_artistids` is not equal to `mb_artistid`.

Below it asserts that either both `mb_artistid` and `mb_artistids` are empty, or `mb_artistid` == `mb_artistids[0]` once the corrections are made by `ensure_consistent_list_fields`.

This test runs for each of the field pairs (`mb_albumartistid` / `mb_albumartistids` and `albumtype` / `albumtypes`).

Reply:
Ah ok so now I understand that as you mentioned below, actually an empty albumtypes is tested here, as we have it in the Discogs plugin. This one I guess:

```
        ("1", []),
```

But wait, the Discogs plugin does not have albumtypes as a list _at all_, what is actually happening if you ask beets to give Item.nonexistent? Will it return None?

Should we add this to the test?

```
        ("1", None),
```

Reply:
See `AlbumInfo.__init__` implementation: if `albumtypes` is not given it's set to an empty list.

Reply:
Ok, damn it. So that's not it.

Reply:
Hopefully your `beet info` outputs will give some insight into what needs to be adjusted in discogs!